File: julia/examples/html_extraction.jl
# html_extraction — build a PDF from Markdown, open it, render the whole
# document to HTML. Exits non-zero on any assertion failure (CI smoke).
using PdfOxide

const MD = "# Hello pdf_oxide\n\nThis is a **Julia** regression example.\n"

pdf = from_markdown(MD)
doc = open_from_bytes(to_bytes(pdf))

html = to_html_all(doc)
println("--- html (all) ---")
println(html)

if !occursin("<", html)
    println("ASSERT FAILED: html does not contain '<'")
    exit(1)
end
if !occursin("pdf_oxide", html)
    println("ASSERT FAILED: html does not contain 'pdf_oxide'")
    exit(1)
end

close!(doc)
println("HTML OK")

File: elixir/examples/html_extraction.exs
# html_extraction — build a PDF from Markdown, open it, render the whole
# document to HTML. Exits non-zero on any assertion failure (CI smoke).
md = "# Hello pdf_oxide\n\nThis is a **Elixir** regression example.\n"

{:ok, pdf} = PdfOxide.from_markdown(md)
{:ok, bytes} = PdfOxide.to_bytes(pdf)
{:ok, doc} = PdfOxide.open_from_bytes(bytes)

{:ok, html} = PdfOxide.to_html_all(doc)
IO.puts("--- html (all) ---")
IO.puts(html)

unless String.contains?(html, "<") do
  IO.puts("ASSERT FAILED: html does not contain '<'")
  System.halt(1)
end

unless String.contains?(html, "pdf_oxide") do
  IO.puts("ASSERT FAILED: html does not contain 'pdf_oxide'")
  System.halt(1)
end

PdfOxide.close(doc)
IO.puts("HTML OK")
